src/utils/CookieUtils.ts

```diff
@@ -33,6 +33,10 @@ export class Cookie {
     }
   }
 
+  private static getHostname() {
+    return location.hostname;
+  }
```

Comment:
How different is this whole file from the one that exists in `coveo.analytics` ? 
Should we just copy paste it ?

What's very important is that the "top level domain" cookie stuff works the same, and that we make sure we write and read in the same manner, with the same expiration, etc.

Reply:
Here's the diff: https://gist.github.com/btaillon/26abe2884f9a972676013d81489a58f6/revisions#diff-a7de55160500e53ee1542467c7f78949d433760e7695b77cfa2b9719c3c9333e

Essentially, they both achieve the same result with the domain.

Reply:
Please doublecheck the domain generation logic, because it's wrong and copied from a random place on the internet (and it was wrong in coveo.analytics.js until we fixed it): 
- Sample input domain `test.me.coveo.com`.
- Expected output: `.coveo.com` (we want the top level domain).

```
host = 'test.me.coveo.com';
domainParts = ['test','me','coveo','com']
// shift
domainParts = [me,'coveo','com']
// assemble using join
domain = '.me.coveo.com'
```
Then it follows with code which attempts to set the cookie on the domain to ensure it's not just '.com' (which presumably is only possible if the input was `blah.com`) and if it is, adds `.` in front of the host (`.blah.com`) and tries the split again, this time removing an empty string 🤦

Reply:
For reference, this is the code block in coveo.analytics.js now:
```
if (host.indexOf('.') === -1) {
    // no "." in a domain - single domain name, it's localhost or something similar
        writeCookie(name, value, expirationDate);
    } else {
        domainParts = host.split('.');
        // we always have at least 2 domain parts
        domain = domainParts[domainParts.length - 2] + '.' + domainParts[domainParts.length - 1];
        writeCookie(name, value, expirationDate, domain);
    }
}
```

Reply:
Leading dots on the domain are ignored by the way: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie

Reply:
🤦  I was on a very outdated branch on the `coveo.analytics` repo when comparing the two.

Fixing this